src/test/java/io/confluent/connect/jdbc/source/integration/BaseConnectorIT.java

```diff
@@ -0,0 +1,228 @@
+  }
+
+  /**
+   * Wait up to {@link #CONNECTOR_STARTUP_DURATION_MS maximum time limit} for the connector with the given
+   * name to start the specified number of tasks.
+   *
+   * @param name     the name of the connector
+   * @param numTasks the minimum number of tasks that are expected
+   * @return the time this method discovered the connector has started, in milliseconds past epoch
+   * @throws InterruptedException if this was interrupted
+   */
+  protected long waitForConnectorToStart(String name, int numTasks) throws InterruptedException {
+    TestUtils.waitForCondition(
+        () -> assertConnectorAndTasksRunning(name, numTasks).orElse(false),
+        CONNECTOR_STARTUP_DURATION_MS,
+        "Connector tasks did not start in time."
+    );
+    return System.currentTimeMillis();
+  }
+
+  /**
+   * Confirm that a connector with an exact number of tasks is running.
+   *
+   * @param connectorName the connector
+   * @param numTasks      the expected number of tasks
+   * @return true if the connector and tasks are in RUNNING state; false otherwise
+   */
+  protected Optional<Boolean> assertConnectorAndTasksRunning(String connectorName, int numTasks) {
+    try {
+      ConnectorStateInfo info = connect.connectorStatus(connectorName);
+      boolean result = info != null
+          && info.tasks().size() == numTasks
+          && info.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+          && info.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
+      return Optional.of(result);
+    } catch (Exception e) {
+      log.warn("Could not check connector state info. Connector might not have started.");
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Create a map of Common connector properties.
+   *
+   * @return : Map of props.
+   */
+  public Map<String, String> getConnectorProps() {
+
+    Map<String, String> props = new HashMap<>();
+    props.put("connector.class", JdbcSourceConnector.class.getName());
+    props.put("tasks.max", MAX_TASKS);
+    props.put("confluent.topic.replication.factor", "1");
+    props.put("confluent.topic.bootstrap.servers", connect.kafka().bootstrapServers());
+
+    return props;
+  }
+
+  protected Connection initializeJdbcConnection() throws SQLException {
+    return DriverManager.getConnection("jdbc:mysql://localhost:3307/db" , "user", "password");
+  }
+
+  protected void assertRecords(int end, ConsumerRecords<byte[], byte[]> records) {
+    int start = 0;
+    for (ConsumerRecord<byte[], byte[]> record : records) {
+      assertEquals(start, record.offset());
+      Struct valueStruct = ((Struct) (jsonConverter.toConnectData(KAFKA_TOPIC, record.value()).value()));
+      int id = valueStruct.getInt32("id");
+      assertEquals(start++, id);
+    }
+    if(start!=end){
+      throw new AssertionError("Invalid number of records found");
+    }
+  }
+
+  protected void addConnectionProperties(Map<String, String> props) {
```

Comment:
why have two separate methods getConnectorProps() and this one?
we should merge them.